pax-compiler/src/parsing.rs

```diff
@@ -932,120 +965,32 @@ fn parse_settings_from_component_definition_string(pax: &str) -> Option<Vec<Sett
                                     );
                                     let literal_object = selector_block_pairs.next().unwrap();
 
-                                    SettingsBlockElement::SelectorBlock(
+                                    settings.push(SettingsBlockElement::SelectorBlock(
                                         token,
                                         derive_value_definition_from_literal_object_pair(
                                             literal_object,
                                             pax,
                                         ),
-                                    )
+                                    ));
                                 }
                                 Rule::comment => {
-                                    let comment = selector_block.as_str().to_string();
-                                    SettingsBlockElement::Comment(comment)
+                                    let comment =
+                                        top_level_settings_block_entity.as_str().to_string();
+                                    settings.push(SettingsBlockElement::Comment(comment));
```

Comment:
This will mess up comment order for handlers since we will serialize the all comments in the settings section of the block irrespective of where they were.